diff --git b/src/Model/Command/Interaction.php a/src/Model/Command/Interaction.php
index 7a606b3..1578502 100644
@@ -147,8 +151,12 @@ class Interaction extends AbstractCommand implements UserAwareInterface
         $this->context = $context;
     }

+    /**
+     * @param int $timestamp
+     */
     protected function setTimestamp($timestamp)
     {
+        Assertion::integer($timestamp);
         Assertion::greaterThan($timestamp, 0);
         $this->timestamp = $timestamp;
     }
diff --git b/src/Model/Command/UserRecommendation.php a/src/Model/Command/UserRecommendation.php
--- b/src/Model/Command/UserRecommendation.php
+++ a/src/Model/Command/UserRecommendation.php
@@ -166,10 +166,11 @@
      * By default user won't see any items, that it has visited (and we have recorded DetailView  interaction.)
      * If you want to circumvent this, and get recommendations including the ones, that the user has already visited,
      * you can set the "seen" allowance here.
-     * @param mixed $seen
+     * @param bool $seen
      */
     public function setAllowSeen($seen)
     {
+        Assertion::boolean($seen);
         $this->allowSeen = $seen;

         return $this;
@@ -210,11 +211,12 @@
      * Set number of requested recommendations. The real number of recommended items could be lower or even zero when
      * there are no items relevant for the user.
      *
-     * @param mixed $count
+     * @param int $count
      * @return $this
      */
     public function setCount($count)
     {
+        Assertion::integer($count);
         Assertion::greaterThan($count, 0);
         $this->count = $count;

@@ -229,6 +231,7 @@
      */
     public function setRotationRate($rotationRate)
     {
+        Assertion::float($rotationRate);
         Assertion::between($rotationRate, 0, 1);
         $this->rotationRate = $rotationRate;

@@ -244,6 +247,7 @@
      */
     public function setRotationTime($rotationTime)
     {
+        Assertion::integer($rotationTime);
         Assertion::greaterOrEqualThan($rotationTime, 0);
         $this->rotationTime = $rotationTime;

diff --git b/src/RequestBuilder/RequestBuilderFactory.php a/src/RequestBuilder/RequestBuilderFactory.php
index f410a97..2f6ccf2 100644
--- a/src/RequestBuilder/RequestBuilderFactory.php
+++ b/src/RequestBuilder/RequestBuilderFactory.php
@@ -29,6 +29,8 @@ class RequestBuilderFactory

     /**
      * Define new properties into the database. Those properties will be created and subsequently accepted by Matej.
+     *
+     * @return ItemPropertiesSetupRequestBuilder
      */
     public function setupItemProperties()
     {
@@ -41,6 +43,8 @@ class RequestBuilderFactory
     /**
      * Added item properties will be IRREVERSIBLY removed from all items in the database and the item property will
      * from now be rejected by Matej.
+     *
+     * @return ItemPropertiesSetupRequestBuilder
      */
     public function deleteItemProperties()
     {
@@ -50,6 +54,9 @@ class RequestBuilderFactory
         return $requestBuilder;
     }

+    /**
+     * @return EventsRequestBuilder
+     */
     public function events()
     {
         $requestBuilder = new EventsRequestBuilder();
@@ -58,6 +65,9 @@ class RequestBuilderFactory
         return $requestBuilder;
     }

+    /**
+     * @return CampaignRequestBuilder
+     */
     public function campaign()
     {
         $requestBuilder = new CampaignRequestBuilder();
@@ -66,6 +76,10 @@ class RequestBuilderFactory
         return $requestBuilder;
     }

+    /**
+     * @param Sorting $sorting
+     * @return SortingRequestBuilder
+     */
     public function sorting(Sorting $sorting)
     {
         $requestBuilder = new SortingRequestBuilder($sorting);
@@ -74,6 +88,10 @@ class RequestBuilderFactory
         return $requestBuilder;
     }

+    /**
+     * @param UserRecommendation $recommendation
+     * @return RecommendationRequestBuilder
+     */
     public function recommendation(UserRecommendation $recommendation)
     {
         $requestBuilder = new RecommendationRequestBuilder($recommendation);
@@ -82,6 +100,9 @@ class RequestBuilderFactory
         return $requestBuilder;
     }

+    /**
+     * @return ForgetRequestBuilder
+     */
     public function forget()
     {
         $requestBuilder = new ForgetRequestBuilder();
@@ -90,6 +111,9 @@ class RequestBuilderFactory
         return $requestBuilder;
     }

+    /**
+     * @return ResetDatabaseRequestBuilder
+     */
     public function resetDatabase()
     {
         $requestBuilder = new ResetDatabaseRequestBuilder();
